solana/indexer/common/pg_notify.go

```diff
@@ -0,0 +1,69 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"api.audius.co/database"
+	"github.com/jackc/pgx/v5/pgconn"
+	"go.uber.org/zap"
+)
+
+type notificationCallback func(ctx context.Context, notification *pgconn.Notification)
+
+// Listens for a notification and fires a callback when one is received.
+// The function spawns a goroutine to listen for notifications, so it returns
+// immediately. The caller should ensure the context is cancelled when they want
+// to stop listening and wait indefinitely to listen.
+func WatchPgNotification(ctx context.Context, pool database.DbPool, notification string, callback notificationCallback, logger *zap.Logger) error {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+
+	childLogger := logger.With(zap.String("notification", notification))
+
+	conn, err := pool.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to acquire database connection: %w", err)
+	}
+
+	rawConn := conn.Conn()
+	_, err = rawConn.Exec(ctx, fmt.Sprintf(`LISTEN %s`, notification))
+	if err != nil {
+		return fmt.Errorf("failed to listen for %s changes: %w", notification, err)
+	}
+
+	go func() {
```

Comment:
I used pgxlisten for this in the comms rpc processor: https://github.com/AudiusProject/api/blob/bae052404d13034ea227bd074a684ea0652e0562/api/comms/rpc_processor.go#L410

Any reason to prefer the raw exec method here?

Reply:
Oooh wasn't aware of that, it looks nice. Will update to that